provider/win_driver_utils.py

```diff
@@ -194,30 +194,135 @@ def copy_file_to_samepath(session, test, params):
                        " the detailed info:\n%s." % output)
 
 
-def memory_leak_check(vm, test, params):
+def enable_driver(vm, test, cmd, device_id):
     """
-    In order to let the driver verifier to catch memory leaks, driver
-    should be unloaded after driver function.Note that if want to use
-    this function, driver verifier should be enabled before driver function.
+    Enable driver.
+
+    :param vm: vm object
+    :param test: kvm test object
+    :param cmd: Driver enable
+    :param device_id: Device id in windows guest
+    """
+    session = vm.wait_for_login()
+    cmd = cmd.replace("DEVICE_ID", device_id)
+    cmd = utils_misc.set_winutils_letter(session, cmd)
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        test.fail("failed to enable driver, %s" % output)
+
+
+def disable_driver(vm, test, cmd, device_id):
+    """
+    Disable driver.
+
+    :param vm: vm object
+    :param test: kvm test object
+    :param cmd: Driver disable command
+    :param device_id: Device id in windows guest
+    """
+    session = vm.wait_for_login()
+    cmd = cmd.replace("DEVICE_ID", device_id)
+    cmd = utils_misc.set_winutils_letter(session, cmd)
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        if "reboot" in output:
+            vm.reboot()
+        else:
+            test.fail("failed to disable driver, %s" % output)
+
+
+def get_device_id(vm, test, params):
+    """
+    Get device id from guest.
 
     :param vm: vm object
     :param test: kvm test object
     :param params: the dict used for parameters
     """
+    device_name = params["device_name"]
+    device_hwid = params["device_hwid"]
+
+    session = vm.wait_for_login()
+    output = _pnpdrv_info(session, device_name, ["DeviceID"])
+    # workaround for viostor/vioscsi to get data device id
+    device_id = output[0]
+    device_id = '&'.join(device_id.split('&amp;'))
+    find_devices = False
+    for hwid in device_hwid.split():
+        hwid = hwid.split('"')[1]
+        if hwid in device_id:
+            find_devices = True
+    if not find_devices:
+        test.fail("Didn't find driver info from guest %s" % output)
+    return device_id
+
+
+def load_driver(vm, test, params, load_method='enable'):
+    """
+    Load driver.
+
+    :param vm: vm object
+    :param test: kvm test object
+    :param params: the dict used for parameters
+    :param load_method: Load driver method
+    """
+    session = vm.wait_for_login()
     driver_name = params["driver_name"]
     devcon_path = params["devcon_path"]
```

Comment:
These do not have to be exist when you choose 'enable', so better move them to the `if load_method != 'enable'` block. Also for below `media_type` and `device_hwid`.

Reply:
> These do not have to be exist when you choose 'enable', so better move them to the `if load_method != 'enable'` block. Also for below `media_type` and `device_hwid`.

@vivianQizhu  updated, thanks